video/out/wayland_common.c

```diff
@@ -2192,6 +2404,59 @@ static void remove_seat(struct vo_wayland_seat *seat)
     return;
 }
 
+static void reset_color_management(struct vo_wayland_state *wl)
+{
+    if (!wl->color_surface)
+        return;
+
+    if (wl->image_creator_params)
+        xx_image_description_creator_params_v4_destroy(wl->image_creator_params);
+
+    if (wl->image_description) {
+        xx_color_management_surface_v4_unset_image_description(wl->color_surface);
+        xx_image_description_v4_destroy(wl->image_description);
+        wl->image_description = NULL;
+    }
+
+    wl->image_creator_params = xx_color_manager_v4_new_parametric_creator(wl->color_manager);
+    wl->unsupported_colorspace = false;
+}
+
+static void set_color_representation(struct vo_wayland_state *wl)
+{
+    if (!wl->color_representation)
+        return;
+
+    struct pl_color_repr repr = wl->vo->params->repr;
```

Comment:
Why `vo->params`? I think for dmabuf_wayland `target_params` should be initialized with `params` values. This will be consistent with other VOs, like gpu which can do processing on its own.

Reply:
I'm honestly really unsure if we should be using `params` or `target_params` with this protocol.

Reply:
Well, we need to signal what we send to compositor, hence the `target_params`. If it is not working as expected for some reason there are different issues.

In most nominal case we want `params` to be equal to `target_params`, but is is not hard rule, as we can still have some processing in VO like vo_gpu/vo_gpu_next and this needs to be reflected in output parameters that are sent to compositor.

Reply:
It is done now.

Reply:
Thanks. One more nitpick, probably would be good to separate target_params. If anything to set `.w` and `.h` to real output values and maybe `.rotate`. The rest should be fine as we don't do any transformations except that.

Reply:
For `vo_dmabuf_wayland` you mean? Should be OK as is. I don't anticipate using anything other than `color` and `repr` from the target_params.

Reply:
It's more like cosmetic issue, it will show wrong res in stats.lua

Reply:
Ah I forgot stats.lua shows this. Fair point.

Reply:
Should be fixed now.